core/templates/pages/exploration-player-page/services/question-player-engine.service.spec.ts

```diff
@@ -0,0 +1,1018 @@
+            hint_1: {}
+          }
+        }
+      },
+      question_state_data_schema_version: 45,
+      language_code: 'en',
+      version: 1,
+      linked_skill_ids: [],
+      inapplicable_skill_misconception_ids: [],
+    }];
+  });
+
+  beforeEach(() => {
+    alertsService = TestBed.inject(AlertsService);
+    answerClassificationService = TestBed.inject(AnswerClassificationService);
+    contextService = TestBed.inject(ContextService);
+    expressionInterpolationService =
+      TestBed.inject(ExpressionInterpolationService);
+    questionObjectFactory = TestBed.inject(QuestionObjectFactory);
+    readOnlyExplorationBackendApiService = TestBed.inject(
+      ReadOnlyExplorationBackendApiService);
+    questionPlayerEngineService = TestBed.inject(QuestionPlayerEngineService);
+    urlService = TestBed.inject(UrlService);
+  });
+
+  it('should load questions on initialization', () => {
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
+    let exploration = {
+      version: '1'
+    } as unknown as FetchExplorationBackendResponse;
+
+    let quesitonPlayerSpy =
+      spyOn(contextService, 'setQuestionPlayerIsOpen').and.returnValue(null);
+    let explorationIdSpy =
+      spyOn(contextService, 'getExplorationId').and.returnValue('id1');
+    let questionPlayerModeSpy =
+      spyOn(contextService, 'isInQuestionPlayerMode').and.returnValue(false);
+    let versionSpy =
+      spyOn(urlService, 'getExplorationVersionFromUrl').and.returnValue(1);
+    let readOnlySpy =
+      spyOn(readOnlyExplorationBackendApiService, 'loadExplorationAsync')
+        .and.returnValue(Promise.resolve(exploration));
+
+    questionPlayerEngineService.init(
+      multipleQuestionBackendDict, successHandler, failHandler);
+
+    expect(quesitonPlayerSpy).toHaveBeenCalled();
+    expect(explorationIdSpy).toHaveBeenCalled();
+    expect(questionPlayerModeSpy).toHaveBeenCalled();
+    expect(versionSpy).toHaveBeenCalled();
+    expect(readOnlySpy).toHaveBeenCalled();
+    expect(successHandler).toHaveBeenCalled();
+    expect(failHandler).not.toHaveBeenCalled();
+
+    // eslint-disable-next-line dot-notation
+    expect(questionPlayerEngineService['explorationId']).toBe('id1');
+    // eslint-disable-next-line dot-notation
+    expect(questionPlayerEngineService['questionPlayerMode']).toBe(false);
+    // eslint-disable-next-line dot-notation
+    expect(questionPlayerEngineService['version']).toBe(1);
+    // eslint-disable-next-line dot-notation
+    expect(questionPlayerEngineService['questions'].length).toBe(3);
+  });
+
+  it('should record if new cards are added', () =>{
+    // eslint-disable-next-line dot-notation
+    questionPlayerEngineService['nextIndex'] = 1;
+
+    questionPlayerEngineService.recordNewCardAdded();
+
+    // eslint-disable-next-line dot-notation
+    expect(questionPlayerEngineService['currentIndex']).toBe(1);
+  });
+
+  it('should get the current question', () =>{
```

Comment:
Please describe the behaviour more clearly. getCurrentQuestion() seems to be the thing being tested here, so you should have a describe block for that. Then, you can say it "should return the current question". On the other hand, if you want the describe to be about the engine service, then you would say that it returns the current question rather than gets it.

Also, you might want to change the definition of what is current and ensure that a new question gets returned after that when getCurrentQuestion() is called. Think about what can go wrong and try to test that. The current test is very minimal.

Reply:
Added extra check.